code/game/objects/items/weapons/RCD.dm

```diff
@@ -580,31 +566,30 @@
 	if(!is_type_in_list(A, allowed_targets))
 		return FALSE
 
-	switch(mode)
-		if(MODE_TURF)
-			. = mode_turf(A, user)
-		if(MODE_AIRLOCK)
-			. = mode_airlock(A, user)
-		if(MODE_DECON)
-			. = mode_decon(A, user)
-		if(MODE_WINDOW)
-			. = mode_window(A, user)
-		else
-			to_chat(user, "ERROR: RCD in MODE: [mode] attempted use by [user]. Send this text #coderbus or an admin.")
-			. = 0
+	for(var/act_type in subtypesof(/datum/rcd_act))
+		var/datum/rcd_act/act = new act_type
+		if(act.can_act(A, src, user))
+			. = act.try_act(A, src, user)
+			update_icon(UPDATE_OVERLAYS)
+			SStgui.update_uis(src)
+			return
+
+	if(mode == MODE_DECON)
+		to_chat(user, "<span class='warning'>You can't deconstruct that!</span>")
+	else
+		to_chat(user, "<span class='warning'>Location unsuitable for construction.</span>")
```

Comment:
```suggestion
	for(var/act_type in subtypesof(/datum/rcd_act))
		var/datum/rcd_act/act = new act_type
		if(act.can_act(A, src, user))
			. = act.try_act(A, src, user)
			update_icon(UPDATE_OVERLAYS)
			SStgui.update_uis(src)
			return

	if(mode == MODE_DECON)
		to_chat(user, "<span class='warning'>You can't deconstruct that!</span>")
	else
		to_chat(user, "<span class='warning'>Location unsuitable for construction.</span>")
```
Can you refactor this to store a list of rcd datums and maybe make it static? Don't want overtime due to this, or weird byondisms